src/Compilers/Core/Portable/CodeGen/SequencePointList.cs

```diff
@@ -124,6 +124,35 @@ public ImmutableArray<Cci.SequencePoint> GetSequencePoints(DebugDocumentProvider
                 current = current._next;
             }
 
+            // Find the document for the first non-hidden sequence point (issue #4370)
+            current = this;
+            while (current != null)
```

Comment:
Actually, I don't think you need to set them in the method. I'd suggest pattern:
FileLinePositionSpan firstNonHiddenSpan;
```if (!TryFindFirstNonHiddenSequencePoint(out firstNonHiddenSpan)) { return ImmutableArray<SequencePoint>.Empty; }```

and then you can directly use ```firstNonHiddenSpan``` instead of the lastXxx variables.